crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -620,6 +620,79 @@ mod private {
             Ok(Self { room: room_id, store, events, waited_for_initial_prev_token: false })
         }
 
+        /// Load more events backwards if the last chunk is **not** a gap.
+        ///
+        /// Return `Ok(Some((events, reached_start), _))` if events have been
+        /// inserted, `Ok(None)` if a gap has been inserted or if the
+        /// store is disabled.
+        #[must_use = "Updates as `VectorDiff` must probably be propagated via `RoomEventCacheUpdate`"]
+        pub(in super::super) async fn load_more_events_backwards(
+            &mut self,
+        ) -> Result<
+            Option<((Vec<TimelineEvent>, bool), Vec<VectorDiff<TimelineEvent>>)>,
+            EventCacheError,
+        > {
+            // ⚠️ Propagate all pending updates before we do anything.
+            self.propagate_changes().await?;
+
+            let Some(store) = self.store.get() else {
+                // No store: no events to insert.
+                return Ok(None);
+            };
+
+            let first_chunk = self
+                .events
+                .chunks()
+                .next()
+                // SAFETY: A `LinkedChunk` is never empty, it always contains at least one chunk.
+                .expect("The `LinkedChunk` in `RoomEvents` cannot be empty");
+
+            // Because `first_chunk` is `not `Send`, get these information before the
+            // `.await` point, so that this `Future` can implement `Send`.
+            let first_chunk_is_gap = first_chunk.is_gap();
```

Comment:
Turns out this can be inlined, as this is used only once, before any await point.

Reply:
True!